src/wallet/wallet2.cpp

```diff
@@ -3557,17 +3624,140 @@ std::string wallet2::make_multisig(const epee::wipeable_string &password,
   return extra_multisig_info;
 }
 
-std::string wallet2::make_multisig(const epee::wipeable_string &password,
-  const std::vector<std::string> &info,
-  uint32_t threshold)
+std::string wallet2::exchange_multisig_keys(const epee::wipeable_string &password,
+  const std::vector<std::string> &info)
+{
+  THROW_WALLET_EXCEPTION_IF(info.empty(),
+    error::wallet_internal_error, "Empty multisig info");
+
+  if (info[0].substr(0, MULTISIG_EXTRA_INFO_MAGIC.size()) == MULTISIG_EXTRA_INFO_MAGIC)
+  {
+    std::vector<crypto::public_key> signers;
+    std::unordered_set<crypto::public_key> pkeys;
+
+    THROW_WALLET_EXCEPTION_IF(!unpack_extra_multisig_info(info, signers, pkeys),
+      error::wallet_internal_error, "Bad extra multisig info");
+
+    return exchange_multisig_keys(password, pkeys, signers);
+  }
+
+  THROW_WALLET_EXCEPTION_IF(false,
+    error::wallet_internal_error, "Unsupported info string");
+  return {};
```

Comment:
Looks a lot simpler to do that in the test above

Reply:
You totally right here :)